src/data_distribution/first_touch_distribution.hpp

```diff
@@ -69,14 +77,46 @@ namespace argo {
 				/**
 				 * @brief perform the necessary directory actions
 				 * @param addr address in the global address space
+				 * @param do_first_touch attempt to first touch if true
 				 */
-				static void update_dirs (const std::size_t& addr);
+				static void update_dirs (const std::size_t& addr, bool do_first_touch = true);
 				/** @brief protects the owners directory */
 				std::mutex owners_mutex;
 
 			public:
+				virtual node_id_t peek_homenode(char* const ptr) {
+					std::size_t page_info[3];
```

Comment:
We should introduce symbolic constants (named macros) for these 3's that appear in this line and in some other ones below.